src/ruby/puppetserver-lib/puppet/server/http_response.rb

```diff
@@ -0,0 +1,52 @@
+require 'puppet/http/response'
+
+class Puppet::Server::HttpResponse < Puppet::HTTP::Response
+  attr_reader :url
+
+  def initialize(java_response, url)
+    @java_response = java_response
+    @url = url
+  end
+
+  def code
+    @java_response.get_status
+  end
+
+  def reason
+    @java_response.get_reason_phrase
+  end
+
+  def body
+    @java_response.get_body
+  end
+
+  def read_body
+    yield body
+  end
```

Comment:
It is not used right now, but is a placeholder for an API for streaming the response body. In a real implementation, it would return chunks of the body one by one, ferrying them from the server to the client. But I talked to Josh and we decided that implementation is out of scope right now, given that the [clojure client](https://github.com/puppetlabs/clj-http-client) we're wrapping doesn't currently implement streaming. It's possible, the underlying java library supports it, just more work than we need right now, given nothing in Puppet Server uses it at the moment.

Reply:
so if this were to be called right now, what would it do? Is a block required or optional (the [puppet implementation](https://github.com/puppetlabs/puppet/blob/main/lib/puppet/http/response.rb#L75) does a check)? I guess I never really got my head wrapped around `yield`..

Reply:
Yeah a block is required. Ruby will error if you call this without it now, when it tries to yield. I can give it a more explicit error like we do in Puppet if we think that's better.

Reply:
Might be nice for consistency's sake, but I don't care too much if Ruby's error is clear.

Reply:
I put in the same error Puppet has.